debian/usr.bin.snap-confine

```diff
@@ -112,6 +114,15 @@
     mount options=(rw bind) /snap/ubuntu-core/*/libx32/ -> /libx32/,
     mount options=(rw bind) /snap/ubuntu-core/*/lib64/ -> /lib64/,
     mount options=(rw bind) /snap/ubuntu-core/*/usr/ -> /usr/,
+    mount options=(rw bind) /snap/ubuntu-core/*/etc/alternatives/ -> /etc/alternatives/,
+
+    # This is a somewhat of a blank check for mount profiles.  Mount profiles
+    # are written by the trusted snapd to a root-writable location and
+    # processed by snap-confine. Regardless of configuration options and
+    # runtime factors they are processed after pivot_root (on classic systems)
+    # so snap.rootfs is not a factor.
+    mount options=(rw bind) /** -> /**,
```

Comment:
As for it happening after the pivot_root, that is assuming that the code is correct before the pivot_root and can't be attacked there. The apparmor profile is attempting to limit the functionality to precisely what the launcher needs, no more, no less. The profile already grants a lot because of the nature of the launcher, but a carte blanche rule like this will provide easier privilege escalation. For example, because of the existing profile, the recent execl bug was considerably harder to expoit for privilege escalation. We want to make sure we are continuing to do this.